exercises/custom-set/.meta/gen.go

```diff
@@ -108,74 +95,100 @@ type binaryOp struct {
 // all converted to strings here.  It just seemed like strings would make
 // a better and more practical example.
 var tmpl = `
-{{/* nested templates for repeated stuff */}}
-
-{{define "binaryBool"}} = []binBoolCase{
-{{range .}}{ // {{.Description}}
-	{{strs .Set1 | printf "%#v"}},
-	{{strs .Set2 | printf "%#v"}},
-	{{.Expected}},
-},
-{{end}}}{{end}}
-
-{{define "binaryOp"}} = []binOpCase{
-{{range .}}{ // {{.Description}}
-	{{strs .Set1 | printf "%#v"}},
-	{{strs .Set2 | printf "%#v"}},
-	{{strs .Expected | printf "%#v"}},
-},
-{{end}}}{{end}}
-
-{{define "eleOp"}} = []eleOpCase{
-{{range .}}{ // {{.Description}}
-	{{strs .Set | printf "%#v"}},
-	{{str .Element | printf "%q"}},
-	{{strs .Expected | printf "%#v"}},
-},
-{{end}}}{{end}}
 
 {{/* begin template body */}}
 package stringset
 
-// Source: {{.Ori}}
-{{if .Commit}}// Commit: {{.Commit}}
-{{end}}
+{{.Header}}
 
-// {{.J.IsEmpty.Description}}
-var emptyCases = []unaryBoolCase{
-{{range .J.IsEmpty.Cases}}{ // {{.Description}}
+{{with .J.Groups}}
+    // {{ .GroupComment "empty"}}
+{{end}} var emptyCases = []unaryBoolCase{ {{range .J.Groups}} {{range .Cases}}
+{{if .PropertyMatch "empty"}} { // {{.Description}}
 	{{strs .Set | printf "%#v"}},
 	{{.Expected}},
 },
-{{end}}}
+{{- end}}{{end}}{{end}} }
+
 
-// {{.J.Contains.Description}}
-var hasCases = []eleBoolCase{
-{{range .J.Contains.Cases}}{ // {{.Description}}
+{{with .J.Groups}}
+    // {{ .GroupComment "contains"}}
+{{end}} var hasCases = []eleBoolCase{ {{range .J.Groups}} {{range .Cases}}
+{{if .PropertyMatch "contains"}} { // {{.Description}}
 	{{strs .Set | printf "%#v"}},
 	{{str .Element | printf "%q"}},
 	{{.Expected}},
 },
-{{end}}}
+{{- end}}{{end}}{{end}} }
 
-// {{.J.Subset.Description}}
-var subsetCases{{template "binaryBool" .J.Subset.Cases}}
+{{with .J.Groups}}
+    // {{ .GroupComment "subset"}}
+{{end}} var subsetCases = []binBoolCase{ {{range .J.Groups}} {{range .Cases}}
+{{if .PropertyMatch "subset"}} { // {{.Description}}
```

Comment:
My attempts to do exactly that have not met with success yet.  I have certainly tried to do it, since there is a nice pattern there.  I will study it more and see.